hybrid/compartment_examples/inter_comp_call/secure/shared.S

```diff
@@ -20,10 +18,8 @@
 executive_switch:
     mov      c29, c0
     mov      x0, #0
+    // Clean up x20 after returning from compartment switch
+    mov      x13, #0
```

Comment:
This is the `leak_x20` argument of `switch_compartment`. I don't like this approach as much as the previous one (where we include the appropriate `clean` implementation based on a flag) because I don't think the compartments should have the power to select a `clean` implementation.

Reply:
I agree that it makes the point that the issue can be more subtle and outside the compartments' hands, as we can easily imagine the compartment being user code, and a the transition code being either in a library, or perhaps even the kernel.